container_api/src/static_file_server.rs

```diff
@@ -0,0 +1,123 @@
+use config::{UiBundleConfiguration, UiInterfaceConfiguration};
+use error::HolochainResult;
+use hyper::{
+    rt::{self, Future},
+    server::Server,
+    Body, Request,
+};
+use std::{io::Error, thread};
+// use tokio::runtime::Runtime;
+use hyper_staticfile::{Static, StaticFuture};
+use tokio::prelude::future;
+
+pub fn notify(msg: String) {
+    println!("{}", msg);
+}
+
+/// Hyper `Service` implementation that serves all requests.
+struct StaticService {
+    static_: Static,
+}
+
+impl StaticService {
+    fn new(path: &String) -> Self {
+        StaticService {
+            static_: Static::new(path),
+        }
+    }
+}
+
+impl hyper::service::Service for StaticService {
+    type ReqBody = Body;
+    type ResBody = Body;
+    type Error = Error;
+    type Future = StaticFuture<Body>;
+
+    fn call(&mut self, req: Request<Body>) -> StaticFuture<Body> {
+        self.static_.serve(req)
+    }
+}
+
+pub struct StaticServer {
+    #[allow(dead_code)]
+    shutdown_signal: Option<futures::channel::oneshot::Sender<()>>,
+    config: UiInterfaceConfiguration,
+    #[allow(dead_code)]
+    bundle_config: UiBundleConfiguration,
+    running: bool,
+}
+
+impl StaticServer {
+    pub fn from_configs(
+        bundle_config: UiBundleConfiguration,
+        config: UiInterfaceConfiguration,
+    ) -> Self {
+        StaticServer {
+            shutdown_signal: None,
+            config,
+            bundle_config,
+            running: false,
+        }
+    }
+
+    pub fn start(&mut self) -> HolochainResult<()> {
+        let addr = ([127, 0, 0, 1], self.config.port).into();
+
+        // let (tx, rx) = futures::channel::oneshot::channel::<()>();
+        // self.shutdown_signal = Some(tx);
+        let static_path = self.bundle_config.root_dir.to_owned();
+
+        println!(
+            "About to serve path \"{}\" at http://{}",
+            &self.bundle_config.root_dir, &addr
+        );
+        self.running = true;
+
+        thread::spawn(move || {
+            let server = Server::bind(&addr)
+                .serve(move || future::ok::<_, Error>(StaticService::new(&static_path)))
+                // .with_graceful_shutdown(rx)
+                .map_err(|e| eprintln!("server error: {}", e));
+
+            println!("Listening on http://{}", addr);
+            rt::run(server)
+        });
+        Ok(())
+    }
+
+    // pub fn stop(&mut self) -> HolochainResult<()> {
```

Comment:
This is the only part that needs to be finished. I guess this can wait until UI bundle CRUD interface though